pyQuARC/code/checker.py

```diff
@@ -157,18 +158,18 @@ def _check_dependencies_validity(self, dependencies, field_dict):
                 return False
         return True
 
-    def _run_func(self, func, rule, rule_id, metadata_content, result_dict):
+    def _run_func(self, func, check, rule_id, metadata_content, result_dict):
         """
         Run the check function for `rule_id` and update `result_dict`
         """
-        dependencies = rule.get("dependencies", [])
         rule_mapping = self.rules_override.get(
             rule_id
         ) or self.rule_mapping.get(rule_id)
         external_data = rule_mapping.get("data", [])
         list_of_fields_to_apply = \
             rule_mapping.get("fields_to_apply").get(self.metadata_format, {})
         for field_dict in list_of_fields_to_apply:
+            dependencies = self.scheduler.get_all_dependencies(rule_id, check, field_dict)
```

Comment:
Since dependencies may be different for different fields (see `rule_mapping.json` for an example), we want to get dependencies that only pertain to a certain field (thus, `field_dict` is passed